src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -402,20 +404,33 @@
           json-fields           (filter #(isa? (:base-type %) :type/JSON) table-fields)]
       (if (nil? (seq json-fields))
         #{}
-        (binding [hx/*honey-sql-version* (sql.qp/honey-sql-version driver)]
-          (let [json-field-names (mapv #(apply hx/identifier :field (into table-identifier-info [(:name %)])) json-fields)
-                table-identifier (apply hx/identifier :table table-identifier-info)
-                sql-args         (sql.qp/format-honeysql driver {:select (mapv sql.qp/maybe-wrap-unaliased-expr json-field-names)
-                                                                 :from   [(sql.qp/maybe-wrap-unaliased-expr table-identifier)]
-                                                                 :limit  metadata-queries/nested-field-sample-limit})
-                query            (jdbc/reducible-query spec sql-args {:identifiers identity})
-                field-types      (transduce describe-json-xform describe-json-rf query)
-                fields           (field-types->fields field-types)]
-            (if (> (count fields) max-nested-field-columns)
-              (do
-                (log/warn
-                 (format
-                  "More nested field columns detected than maximum. Limiting the number of nested field columns to %d."
-                  max-nested-field-columns))
-                (set (take max-nested-field-columns fields)))
-              fields)))))))
+        (let [existing-fields-by-name (m/index-by :name (t2/select 'Field :table_id (u/the-id table)))
+              unfold-json-fields      (filter (fn [field]
+                                                ;; unfold json if the field has json_unfolding = true
+                                                ;; don't unfold json if the field has json_unfolding = false
+                                                ;; otherwise if the field doesn't exist
+                                                ;; use the database default
+                                                (let [existing-field (existing-fields-by-name (:name field))]
+                                                  (or (:json_unfolding existing-field)
```

Comment:
do we have tests for this? I see we have `describe-nested-field-columns-test` in `describe_table_test.clj` but didn't see we updating that test.